src/update-feeds/update-transactions.ts

```diff
@@ -0,0 +1,169 @@
+}
+
+export const updateFeeds = async (
+  api3ServerV1: Api3ServerV1,
+  gasPrice: ethers.BigNumber,
+  updateableDapis: UpdateableDapi[]
+) => {
+  const state = getState();
+  const {
+    config: { sponsorWalletMnemonic },
+  } = state;
+
+  // Update all of the dAPIs in parallel.
+  await Promise.all(
+    updateableDapis.map(async (dapi) => {
+      const { dapiInfo, updateableBeacons } = dapi;
+      // TODO: dapiInfo.dataFeed as of now is the encoded data feed information.
+      const { dapiName, dataFeed: dataFeedId } = dapiInfo;
+
+      await logger.runWithContext({ dapiName, dataFeedId }, async () => {
+        const goUpdate = await go(async () => {
+          logger.debug('Updating dAPI');
+
+          // Create calldata for all beacons of the particular data feed the dAPI points to.
+          const beaconUpdateCalls = updateableBeacons.map((beacon) => {
+            const { signedData } = beacon;
+
+            return api3ServerV1.interface.encodeFunctionData('updateBeaconWithSignedData', [
+              signedData.airnode,
+              signedData.templateId,
+              signedData.timestamp,
+              signedData.encodedValue,
+              signedData.signature,
+            ]);
+          });
+
+          // If there are multiple beacons in the data feed it's a beacons set which we need to update as well.
+          const dataFeedUpdateCalldatas =
+            beaconUpdateCalls.length > 1
+              ? [
+                  ...beaconUpdateCalls,
+                  api3ServerV1.interface.encodeFunctionData('updateBeaconSetWithBeacons', [
+                    updateableBeacons.map(({ beaconId }) => beaconId),
+                  ]),
+                ]
+              : beaconUpdateCalls;
+
+          logger.debug('Estimating gas limit');
+          const gasLimit = await estimateMulticallGasLimit(
+            api3ServerV1,
+            dataFeedUpdateCalldatas,
+            updateableBeacons.map((beacon) => beacon.beaconId)
+          );
+
+          logger.debug('Deriving sponsor wallet');
+          // TODO: These wallets could be persisted as a performance optimization.
+          const sponsorWallet = deriveSponsorWallet(sponsorWalletMnemonic, dapiName);
+
+          logger.debug('Updating dAPI', { gasPrice, gasLimit });
+          await api3ServerV1
+            // When we add the sponsor wallet (signer) without connecting it to the provider, the provider of the
+            // contract will be set to "null". We need to connect the sponsor wallet to the provider of the contract.
+            .connect(sponsorWallet.connect(api3ServerV1.provider))
+            .tryMulticall(dataFeedUpdateCalldatas, { gasPrice, gasLimit });
+          logger.debug('Successfully updated dAPI');
+        });
+
+        if (!goUpdate.success) {
+          logger.error(`Failed to update a dAPI`, goUpdate.error);
+        }
+      });
+    })
+  );
+};
+
+export const estimateMulticallGasLimit = async (
```

Comment:
Fully based on Airseeker v1 implementation. Later we might do https://github.com/api3dao/airseeker-v2/issues/67 but that needs a discussion and agreement first.

Reply:
Hey, AFAIK in airseeker v1 a configured gas limit is used (gas for calls is not estimated). 
Separately, manager-multisig often has to do multicalls and the gas is sometimes incorrectly estimated (often repeatedly for the same transaction). It's just something we should probably keep in mind; if we do estimates will we have a config parameter for adding headroom?

Reply:
> if we do estimates will we have a config parameter for adding headroom?

There is 10% headroom added when we do estimate of the `multicall`, so I guess that covers the need for `tryMulticall` and also adds this headroom.

Reply:
It's possible I made a mistake, thanks for clarifying :)
The current Airseeker implementation works well, so it must be fine :)